From ce753a0819a45f731bd05c93e948899699acf5f1 Mon Sep 17 00:00:00 2001
From: Tim Blechmann <tim.blechmann@qt.io>
Date: Mon, 6 Oct 2025 12:29:05 +0800
Subject: [PATCH 285/553] xcb: QXcbDrag - avoid excessive calls to
 currentDrag()

The code is easier to read/debug when the result of currentDragObject is
stored in a local variable. This also avoids going through the singleton
initialization code more often than necessary.

Change-Id: I1bfd31fd0f777ff5fa63b1baef7183495c6f2b81
Reviewed-by: Liang Qi <liang.qi@qt.io>
(cherry picked from commit bbc579410a896d461868707b373bef1a5d48573c)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 src/plugins/platforms/xcb/qxcbdrag.cpp | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/src/plugins/platforms/xcb/qxcbdrag.cpp b/src/plugins/platforms/xcb/qxcbdrag.cpp
index 2d35c44c489..8814f91d916 100644
--- a/src/plugins/platforms/xcb/qxcbdrag.cpp
+++ b/src/plugins/platforms/xcb/qxcbdrag.cpp
@@ -995,10 +995,12 @@ void QXcbDrag::handleDrop(QPlatformWindow *, const xcb_client_message_event_t *e
         if (dropData && dropData->hasImage())
             dropData = 0;
     }
+
+    const QDrag *currentDragObject = currentDrag();
     // if we can't find it, then use the data in the drag manager
-    if (currentDrag()) {
+    if (currentDragObject) {
         if (!dropData)
-            dropData = currentDrag()->mimeData();
+            dropData = currentDragObject->mimeData();
         supported_drop_actions = Qt::DropActions(l[4]);
     } else {
         if (!dropData)
@@ -1009,8 +1011,8 @@ void QXcbDrag::handleDrop(QPlatformWindow *, const xcb_client_message_event_t *e
     if (!dropData)
         return;
 
-    auto buttons = currentDrag() ? b : connection()->queryMouseButtons();
-    auto modifiers = currentDrag() ? mods : connection()->keyboard()->queryKeyboardModifiers();
+    auto buttons = currentDragObject ? b : connection()->queryMouseButtons();
+    auto modifiers = currentDragObject ? mods : connection()->keyboard()->queryKeyboardModifiers();
 
     QPlatformDropQtResponse response = QWindowSystemInterface::handleDrop(
                 currentWindow.data(), dropData, currentPosition, supported_drop_actions,
-- 
2.51.2

